src/drivers/legos/driver.rs

```diff
@@ -0,0 +1,402 @@
+                    x: state.l_stick_x,
+                    y: state.l_stick_y,
+                })));
+            }
+            if state.r_stick_x != old_state.r_stick_x || state.r_stick_y != old_state.r_stick_y {
+                events.push(Event::Axis(AxisEvent::RStick(JoyAxisInput {
+                    x: state.r_stick_x,
+                    y: state.r_stick_y,
+                })));
+            }
+
+            if state.a_trigger_l != old_state.a_trigger_l {
+                events.push(Event::Trigger(TriggerEvent::ATriggerL(TriggerInput {
+                    value: state.a_trigger_l,
+                })));
+            }
+            if state.a_trigger_r != old_state.a_trigger_r {
+                events.push(Event::Trigger(TriggerEvent::ATriggerR(TriggerInput {
+                    value: state.a_trigger_r,
+                })));
+            }
+            //TODO: When touchpad firmware is updated to use ABS events, enable this
+            //if state.touch_x != old_state.touch_x
+            //    || state.touch_y != old_state.touch_y
+            //    || state.rpad_touching != old_state.rpad_touching
+            //{
+            //    events.push(Event::Axis(AxisEvent::Touchpad(TouchAxisInput {
+            //        x: state.touch_x,
+            //        y: state.touch_y,
+            //        index: 0,
+            //        is_touching: state.rpad_touching,
+            //    })))
+            //}
+        };
+
+        events
+    }
+
+    fn handle_inertial_report(
+        &mut self,
+        buf: [u8; INERTIAL_PACKET_SIZE],
+    ) -> Result<Vec<Event>, Box<dyn Error + Send + Sync>> {
+        let input_report = InertailInputDataReport::unpack(&buf)?;
+
+        // Print input report for debugging
+        //log::debug!("--- Input report ---");
+        //log::debug!("{input_report}");
+        //log::debug!(" ---- End Report ----");
+
+        let report_type = match input_report.report_id {
+            1 => ReportType::AccelData,
+            2 => ReportType::GyroData,
+            _ => {
+                let _report_id = input_report.report_id;
+                return Err("Unknown report type: {_report_id}".into());
+            }
+        };
+
+        match report_type {
+            ReportType::AccelData => {
+                // Update the state
+                let old_state = self.update_accel_state(input_report);
+                // Translate the state into a stream of input events
+                let events = self.translate_accel_data(old_state);
+                return Ok(events);
+            }
+            ReportType::GyroData => {
+                // Update the state
+                let old_state = self.update_gyro_state(input_report);
+                // Translate the state into a stream of input events
+                let events = self.translate_gyro_data(old_state);
+                return Ok(events);
+            }
+        };
```

Comment:
To fix clippy warnings:

```suggestion
        match report_type {
            ReportType::AccelData => {
                // Update the state
                let old_state = self.update_accel_state(input_report);
                // Translate the state into a stream of input events
                let events = self.translate_accel_data(old_state);
                Ok(events)
            }
            ReportType::GyroData => {
                // Update the state
                let old_state = self.update_gyro_state(input_report);
                // Translate the state into a stream of input events
                let events = self.translate_gyro_data(old_state);
                Ok(events)
            }
        }
```